runtime/src/main/java/com/flipkart/flux/dao/AbstractDAO.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2012-2016, the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.flux.dao;
+
+import com.flipkart.flux.util.HibernateUtil;
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.criterion.Restrictions;
+
+/**
+ * @author shyam.akirala
+ */
+public class AbstractDAO<T> {
```

Comment:
Lets not open sessions or transactions in the dao.
Lets do it using AOP/Decorators (similar to DW UnitOfWork). That way we can keep it at any layer above the DAO layers (or on the DAO layers) that may need to work with multiple DAOs and thus need a session and tx across the whole shebang